mm.c

```diff
@@ -623,6 +654,55 @@ static void test_place_block_into_free_list()
     printf("Test passed.\n\n");
 }
 
+static void test_find_free_block() {
+	printf("Test find_free_block.\n");
+
+	char *free_list_backup[MAX_POWER + 1]; // backing up main_free_list to restore after test
+	for (int i = 0; i <= MAX_POWER; i++) {
+		free_list_backup[i] = GET_FREE_LIST_PTR(i);
+	}
+
+	char **bp;
+
+	test_place_block_into_free_list_helper();
+
+	// Case 1: find by size that is smaller than index
+	bp = find_free_block(2);
+	assert(GET_SIZE(bp) == 40);
+	printf("Case 1 passed.\n");
+
+	// Case 2: find by size that matches smallest block in index
+	bp = find_free_block(33);
+	assert(GET_SIZE(bp) == 40);
```

Comment:
Since we will be taking out the tests later I don't think this is a good idea to have extra macros around.